pallets/ddc-staking/src/lib.rs

```diff
@@ -965,6 +965,34 @@ pub mod pallet {
 		}
 	}
 
+	impl<T: Config> Staker<T, BalanceOf<T>> for Pallet<T> {
+		fn bond_stake_and_serve(
+			stash: T::AccountId,
+			controller: T::AccountId,
+			node: NodePubKey,
+			value: BalanceOf<T>,
+			cluster_id: ClusterId,
+		) -> DispatchResult {
+			Nodes::<T>::insert(&node, &stash);
+			Providers::<T>::insert(&stash, &node);
+			<Bonded<T>>::insert(&stash, &controller);
+			let stash_balance = T::Currency::free_balance(&stash);
+			let value = value.min(stash_balance);
+			Self::deposit_event(Event::<T>::Bonded(stash.clone(), value));
+			let item = StakingLedger {
+				stash: stash.clone(),
+				total: value,
+				active: value,
+				chilling: Default::default(),
+				unlocking: Default::default(),
+			};
+			Self::update_ledger(&controller, &item);
+			Self::do_add_cdn(&stash, cluster_id);
```

Comment:
As `bond_stake_and_serve` function accepts `NodePubKey` of any type (both Storage and CDN), it should be able to process a node of any type. Please add code that handles staking for the Storage node as well

Reply:
it seems like we're constantly having these if/else for Storage/CDN. for now, let's fix it with another if/else, but once we have another node type, let's create a trait and factory returning different implementations for each node type